src/lib/CoreInstance.cc

```diff
@@ -33,16 +33,15 @@ CoreInstance::~CoreInstance() {
 
 void CoreInstance::generateCoreModel(std::string executablePath,
                                      std::vector<std::string> executableArgs) {
-  setSimulationMode();
   createProcess(executablePath, executableArgs);
   // Check to see if either of the instruction or data memory interfaces should
   // be created. Don't create the core if either interface is marked as External
   // as they must be set manually prior to the core's creation.
 
   // Convert Data-Memory's Interface-Type value from a string to
   // simeng::MemInterfaceType
-  std::string dType_string =
-      config_["L1-Data-Memory"]["Interface-Type"].as<std::string>();
+  std::string dType_string;
+  config_["L1-Data-Memory"]["Interface-Type"] >> dType_string;
```

Comment:
The use of SimInfo's `getValue` function seems vacant in this file. Given it exists we should use it to avoid these multi-line value accesses. Apply throughout

Reply:
I fall on Finn's side of the fence here.  The getValue function is easier for me to read and understand than the >> operator.

Reply:
Config getter functionality has been changed to new custom ryml `as` function as discussed offline.